src/deriver/api.py

```diff
@@ -378,10 +378,19 @@ def derive_smiles_gb(self, n_children: int = 100, mut_rate: float = 0.01):
             filter_params = None
 
         for _ in range(n_children):
-            parent_a_smiles, parent_b_smiles = random.sample(self.data.seed_smiles, 2)
-            parent_a, parent_b = [Chem.MolFromSmiles(s) for s in (parent_a_smiles, parent_b_smiles)]
             try:
-                new_child = crossover_gb(parent_a, parent_b)
+                parent_a_smiles, parent_b_smiles = random.sample(self.data.seed_smiles, 2)
+                do_crossover = True
+                new_child = None
+            except ValueError:
```

Comment:
Why rely on a valueerror to detect this? Just check first if there are two seeds with an if statement. No need for try except unless I'm missing something

Reply:
they are functionally equivalent since there is only one case where random sample gets a value error. But an if else is just more clear, so I will change it